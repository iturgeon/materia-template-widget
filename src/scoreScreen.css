// Template Note: Boilerplate styles for any widget
body {
	background: #FFFFFF;
	font-family: Lato, sans-serif;
}
