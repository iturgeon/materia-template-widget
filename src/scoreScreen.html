<!DOCTYPE html>
<html>
	<head>
		<title>Template Widget Score Screen</title>
		<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />

		<!-- STYLESHEETS -->
		<link rel="stylesheet" type="text/css" href="scoreScreen.css">

		<!-- REQUIRED MATERIA JAVASCRIPT -->
		<script src="materia.scorecore.js"></script>

		<!-- YOUR PREREQUISITES -->
		<script src="//cdnjs.cloudflare.com/ajax/libs/react/16.4.2/umd/react.development.js"></script>
		<script src="//cdnjs.cloudflare.com/ajax/libs/react-dom/16.4.2/umd/react-dom.development.js"></script>

		<!-- MAIN PLAYER APP -->
	</head>

	<body>
		<div id="root"></div>
		<script src="scoreScreen.js"></script>
	</body>
</html>
