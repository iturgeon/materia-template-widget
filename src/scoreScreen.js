import React from 'react';
import ReactDOM from 'react-dom';
import ScoreScreen from './scoreScreen-component';

const defaultParams = {
	title: '',
	scoreTable: []
}

let currentState = Object.assign({}, defaultParams)

const updateDisplay = (qset, scoreTable, title) => {
	let newState = {
		title:title,
		scoreTable: scoreTable[0].details[0].table
	}

	currentState = Object.assign({}, currentState, newState)

	ReactDOM.render(
		<ScoreScreen {...currentState} />,
		document.getElementById('root')
	)
}

const materiaCallbacks = {
	start: (instance, qset, scoreTable, isPreview, qsetVersion) => {
		updateDisplay(qset, scoreTable, instance.name)
	},
	update: (qset, scoreTable) => {
		updateDisplay(qset, scoreTable, currentState.title)
	}
}

Materia.ScoreCore.start(materiaCallbacks)
