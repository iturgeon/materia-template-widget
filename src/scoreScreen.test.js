import React from 'react';
import renderer from 'react-test-renderer';

const mockDomRender = jest.fn()
jest.mock('react-dom', () => ({
     render: mockDomRender,
}));

// jest.mock('react-dom', () => ({ReactDom: {render: mockDomRender}}), {virtual: true})
// jest.mock('./react-render', () => mockDomRender, {virtual: true})

const mockScoreScreen = jest.fn()
jest.mock('./scoreScreen-component', () => mockScoreScreen, {virtual: true})

describe('ScoreScreen', function() {

	beforeEach(() => {
		jest.resetAllMocks()
		jest.resetModules()
		global.Materia = {
			ScoreCore: {
				start: jest.fn()
			}
		}
	})

	test('ScoreScreen regesters callbacks', () => {


		require('./scoreScreen')
		expect(Materia.ScoreCore.start).toHaveBeenCalledTimes(1)
		const callbacks = Materia.ScoreCore.start.mock.calls.pop()[0]
		expect(callbacks).toHaveProperty('start', expect.any(Function))
		expect(callbacks).toHaveProperty('update', expect.any(Function))
	});

	test('start renders as expected', () => {
		require('./scoreScreen')
		const start = Materia.ScoreCore.start.mock.calls.pop()[0].start

		const scoreTable = [
			{
				details: [
					{
						table: [
							{
								data: [
									"Mock Question 1",
									"Mock Correct Answer 1",
									"Mock Given Answer 1"
								],
								score: 100,
								symbol: "%"
							},
							{
								data: [
									"Mock Question 2",
									"Mock Correct Answer 2",
									"Mock Given Answer 2"
								],
								score: 0,
								symbol: " points"
							}
						]
					}
				]
			}
		]

		expect(mockDomRender).toHaveBeenCalledTimes(0)
		start({name: 'mock-name'}, {}, scoreTable, false, 1)
		expect(mockDomRender).toHaveBeenCalledTimes(1)
		expect(mockDomRender.mock.calls[0][0]).toMatchSnapshot()
	});

	test('update renders as expected', () => {
		require('./scoreScreen')
		const update = Materia.ScoreCore.start.mock.calls.pop()[0].update

		const scoreTable = [
			{
				details: [
					{
						table: [
							{
								data: [
									"Mock Question 1",
									"Mock Correct Answer 1",
									"Mock Given Answer 1"
								],
								score: 100,
								symbol: "%"
							},
							{
								data: [
									"Mock Question 2",
									"Mock Correct Answer 2",
									"Mock Given Answer 2"
								],
								score: 0,
								symbol: " points"
							}
						]
					}
				]
			}
		]

		expect(mockDomRender).toHaveBeenCalledTimes(0)
		update({}, scoreTable)
		expect(mockDomRender).toHaveBeenCalledTimes(1)
		expect(mockDomRender.mock.calls[0][0]).toMatchSnapshot()
	});

})
