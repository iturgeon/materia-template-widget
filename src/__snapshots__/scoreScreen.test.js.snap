// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScoreScreen start renders as expected 1`] = `
<mockConstructor
  scoreTable={
    Array [
      Object {
        "data": Array [
          "Mock Question 1",
          "Mock Correct Answer 1",
          "Mock Given Answer 1",
        ],
        "score": 100,
        "symbol": "%",
      },
      Object {
        "data": Array [
          "Mock Question 2",
          "Mock Correct Answer 2",
          "Mock Given Answer 2",
        ],
        "score": 0,
        "symbol": " points",
      },
    ]
  }
  title="mock-name"
/>
`;

exports[`ScoreScreen update renders as expected 1`] = `
<mockConstructor
  scoreTable={
    Array [
      Object {
        "data": Array [
          "Mock Question 1",
          "Mock Correct Answer 1",
          "Mock Given Answer 1",
        ],
        "score": 100,
        "symbol": "%",
      },
      Object {
        "data": Array [
          "Mock Question 2",
          "Mock Correct Answer 2",
          "Mock Given Answer 2",
        ],
        "score": 0,
        "symbol": " points",
      },
    ]
  }
  title=""
/>
`;
